### PR TITLE
Add preprocessor to replace HTML images with responsive images

### PR DESCRIPTION
**Pull request by gabrielpoca**
With this change, we introduce a preprocessor that's capable of replacing images with responsive images in HTML. Images that have a `srcset` or a `no-responsive-image` are ignored.

The markdown preprocessor was updated to run the generated HTML through this new preprocessor, making it super simple to use for blogs. This behavior is controlled through a flag.

Extra:

The calls to `ErrorCache` are not async. There's no reason for them to be sync that I can think of. On top of that, it was breaking the tests because the `ErrorCache` process is not running.

### CODE REVIEW

File: lib/still/preprocessor/html_responsive_image.ex [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+    new_content =
+      document
+      |> Floki.find_and_update("img", fn
+        {"img", img_attrs} ->
+          src = find_node_attr(img_attrs, "src")
+          srcset = find_node_attr(img_attrs, "srcset")
+          no_responsive_image = find_node_attr(img_attrs, @no_responsive_image)
+
+          if is_image?(src) && is_nil(srcset) && is_nil(no_responsive_image) do
```
Comment (frm) [inline lib/still/preprocessor/html_responsive_image.ex:33]: Code style: I'd abstract these `if` branches to something more meaningful. At least when doing code reviews, it's a lot harder to reason when interpreting specific conditions in a high-level abstraction function.

**Review by frm — APPROVED** [review]
Just the one comment, it's mostly a recommendation, I'd have you do as you please 😄

File: lib/still/preprocessor/html_responsive_image.ex (hunk 1 above)
Comment (gabrielpoca) [reply lib/still/preprocessor/html_responsive_image.ex]: Updated with a different approach!

**Comment by frm** [comment]
Already approved

**Pull request merged**